### ISSUE TITLE
bug(Datepicker): mat-datepicker-input has aria-haspopup attribute but input field does not trigger any popup

### ISSUE THREAD

**Issue by RiinaKuu** [angular/components#31814]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

`mat-datepicker-input` input element uses `aria-haspopup="dialog"` but input element itself does not trigger any popup. The calendar dialog popup can be triggered only from the `mat-datepicker-toggle` button, which correctly has the `aria-haspopup` attribute. 

This violates [WCAG Success Criterion 4.1.2: Name, Role, Value (Level A)](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html).

### Reproduction

Example link: https://material.angular.dev/components/datepicker/overview

Steps to reproduce:
- Go to the Angular Material link above (I used Chrome)
- Start the screen reader (I used VoiceOver on MacOS)
- Navigate to the first _Basic Datepicker_ example and its input field
- Notice that it is informing that the input has dialog pop up

### Expected Behavior

The input field should not have screen reader information about popup, since it is not triggering anything.

### Actual Behavior

<img width="807" height="292" alt="Image" src="https://github.com/user-attachments/assets/5d1238ef-b945-4330-8216-523524a5fc84" />

### Environment

- Angular:
- CDK/Material: 20.2.1
- Browser(s): Chrome
- Operating System (e.g. Windows, macOS, Ubuntu): macOS

**Comment by crisbeto** [comment]
The input can trigger the popup by pressing alt + down arrow.

**Comment by RiinaKuu** [comment]
I see, it sure opens the popup in that way!
However, **interactive element should only contain one functionality**. It is unclear for screen reader users what is expected from them since there is no single clear role. The input element is both a popup trigger and text edit, so user might not understand the meaning of that element. And since there is a way to open the calendar popup from an actual button, it makes the aria attribute in the input field redundant (in addition to the fact that it is violating the WCAG Understanding Success Criterion).

**Comment by crisbeto** [comment]
I believe the functionality is based on this example from the W3C: https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-datepicker/
